src/application.coffee

```diff
@@ -332,6 +363,122 @@ updateStatus =
 	forceNext: false
 	intervalHandle: null
 
+updateStrategies =
```

Comment:
There's a fair chunk of duplicated code between these strategies, I'm not sure it's particularly avoidable though since it's largely plumbing, except maybe a shared `.catch` handler.  Basically just keep it in mind, although I'm not sure you can do much about it